dynamics/src/cgParametricMomentum.cpp

```diff
@@ -410,195 +384,159 @@ namespace Nextsim {
+    const DGVector<1>& H, const DGVector<1>& A, const DGVector<1>& D);
+template void CGParametricMomentum<1>::prepareIteration(
+    const DGVector<3>& H, const DGVector<3>& A, const DGVector<3>& D);
+template void CGParametricMomentum<1>::prepareIteration(
+    const DGVector<6>& H, const DGVector<6>& A, const DGVector<6>& D);
+template void CGParametricMomentum<2>::prepareIteration(
+    const DGVector<1>& H, const DGVector<1>& A, const DGVector<1>& D);
+template void CGParametricMomentum<2>::prepareIteration(
+    const DGVector<3>& H, const DGVector<3>& A, const DGVector<3>& D);
+template void CGParametricMomentum<2>::prepareIteration(
+    const DGVector<6>& H, const DGVector<6>& A, const DGVector<6>& D);
+
+// --------------------------------------------------
+
+template void CGParametricMomentum<1>::mEVPStep(const VPParameters& params, size_t NT_evp,
+    double alpha, double beta, double dt_adv, const DGVector<1>& H, const DGVector<1>& A);
+template void CGParametricMomentum<1>::mEVPStep(const VPParameters& params, size_t NT_evp,
+    double alpha, double beta, double dt_adv, const DGVector<3>& H, const DGVector<3>& A);
+template void CGParametricMomentum<1>::mEVPStep(const VPParameters& params, size_t NT_evp,
+    double alpha, double beta, double dt_adv, const DGVector<6>& H, const DGVector<6>& A);
+
+template void CGParametricMomentum<2>::mEVPStep(const VPParameters& params, size_t NT_evp,
+    double alpha, double beta, double dt_adv, const DGVector<1>& H, const DGVector<1>& A);
+template void CGParametricMomentum<2>::mEVPStep(const VPParameters& params, size_t NT_evp,
+    double alpha, double beta, double dt_adv, const DGVector<3>& H, const DGVector<3>& A);
+template void CGParametricMomentum<2>::mEVPStep(const VPParameters& params, size_t NT_evp,
+    double alpha, double beta, double dt_adv, const DGVector<6>& H, const DGVector<6>& A);
+
+// --------------------------------------------------
+
+template void CGParametricMomentum<1>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<1>& H, const DGVector<1>& A, DGVector<1>& D);
+template void CGParametricMomentum<1>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<3>& H, const DGVector<3>& A, DGVector<3>& D);
+template void CGParametricMomentum<1>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<6>& H, const DGVector<6>& A, DGVector<6>& D);
+
+template void CGParametricMomentum<2>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<1>& H, const DGVector<1>& A, DGVector<1>& D);
+template void CGParametricMomentum<2>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<3>& H, const DGVector<3>& A, DGVector<3>& D);
+template void CGParametricMomentum<2>::MEBStep(const MEBParameters& params, size_t NT_evp,
+    double dt_adv, const DGVector<6>& H, const DGVector<6>& A, DGVector<6>& D);
```

Comment:
I'm not sure I don't like the original format better. It's rather easier to read than the more compact `clang-format`ted version.